src/mlpack/core/data/image_resize_crop.hpp

```diff
@@ -0,0 +1,163 @@
+/**
+ * @file core/data/image_resize_crop.hpp
+ * @author Omar Shrit
+ *
+ * Image resize and crop functionalities.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+#define MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/stb/stb.hpp>
+
+#include "image_info.hpp"
+
+namespace mlpack {
+namespace data {
+
+#ifndef MLPACK_DISABLE_STB
+
+/**
+ * Image resize/crop interfaces.
+ */
+
+/**
+ * Resize one single image matrix.
+ *
+ * This function should be used if the image is loaded as an armadillo matrix
+ * and the number of cols equal to the Width and the number of rows equal
+ * the Height of the image, or the total number of image pixels is equal to the
+ * number of element in an armadillo matrix.
+ *
+ * @param image The input matrix that contains the image to be resized.
+ * @param info Contains relevant input images information.
+ * @param newWidth The new requested width for the resized image.
+ * @param newHeight The new requested height for the resized image.
+ */
+template<typename eT>
+inline void Resize(arma::Mat<eT>& image, data::ImageInfo& info,
+    const size_t newWidth, const size_t newHeight)
```

Comment:
Where should we document this?  I guess for now we could put it in `doc/user/load_save.md` under the "Image data" section but that's a little awkward.  Maybe we should make a new page for it?  I do definitely want to make sure that we add documentation before we merge this though.  (It shouldn't be hard, you mostly have the documentation already here :))

Reply:
For now let us put it under Load section. Once we have the crop functionality in addition to this we can move it to a new page.
Given that the Load API is going to change as well, then can move all of them to a page called Image Support or something like that.